### PR TITLE
Disable normalization of string vnodes

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Fixes: https://github.com/preactjs/preact/issues/2387

What would happen is it would see the prototype and shift it through normalization, since string vnode.props is the actual string we don't want this. This adds a gaurd to avoid this check.

### CODE REVIEW

File: compat/src/render.js [hunk 1]
```diff
@@ -104,62 +104,66 @@ options.vnode = vnode => {
...
+
+			// Normalize DOM vnode properties.
+			let shouldSanitize, attrs, i;
+			for (i in props) if ((shouldSanitize = CAMEL_PROPS.test(i))) break;
+			if (shouldSanitize) {
+				attrs = vnode.props = {};
+				for (i in props) {
+					attrs[
+						CAMEL_PROPS.test(i)
+							? i.replace(/([A-Z0-9])/, '-$1').toLowerCase()
```
Comment (jviide) [inline compat/src/render.js]: Suggestion: maybe `i.replace(/[A-Z0-9]/, '-$&').toLowerCase()` could save a bit or two?

Comment (JoviDeCroock) [reply compat/src/render.js]: Added, thanks!

File: compat/test/browser/createElement.test.js [hunk 2]
```diff
@@ -46,4 +46,16 @@ describe('compat createElement()', () => {
...
+	it('should not normalize text nodes', () => {
+		String.prototype.capFLetter = function() {
+			return this.charAt(0).toUpperCase() + this.slice(1);
+		};
+		let vnode = <div>hi buddy</div>;
+
+		render(vnode, scratch);
+
+		expect(scratch.innerHTML).to.equal('<div>hi buddy</div>');
+		console.log(scratch);
```
Comment (marvinhagemeister) [inline compat/test/browser/createElement.test.js]: Whoops 🙃

**Comment by jedlikowski** [comment]
@JoviDeCroock hi guys, do you have an ETA when this will be released? Sorry to barge in like that, I have an issue with our widget on a client's website and I hope this will resolve it 👍

**Comment by JoviDeCroock** [comment]
Seems to be released according to our releases log here on GitHub

**Comment by jedlikowski** [comment]
I see the last release 28 days ago (10.3.4) and commits from this PR are on the list of `commits to master since this release`.

**Comment by marvinhagemeister** [comment]
I'll cut one, give me a minute 💯

**Comment by jedlikowski** [comment]
Wow guys, I must say I'm very surprised by your responsiveness 😄 . It's my first time on preact repo and definetely haven't had a similar experience anywhere else

**Comment by JoviDeCroock** [comment]
That’s nice to hear, thank you! 🤗

**Pull request merged**